sources/packages/backend/apps/queue-consumers/src/processors/schedulers/institution-integration/ece-request/ece-process-integration.scheduler.ts

```diff
@@ -0,0 +1,49 @@
+import { InjectQueue, Process, Processor } from "@nestjs/bull";
+import { ECEFileService } from "@sims/integrations/institution-integration/ece-integration";
+import { QueueService } from "@sims/services/queue";
+import { QueueNames } from "@sims/utilities";
+import { InjectLogger, LoggerService } from "@sims/utilities/logger";
+import { Job, Queue } from "bull";
+import { BaseScheduler } from "../../base-scheduler";
+import {
+  GeneratedDateQueueInDTO,
+  ECEProcessResultQueueOutDTO,
+} from "./models/ece.model";
+
+@Processor(QueueNames.ECEProcessIntegration)
+export class ECEProcessIntegrationScheduler extends BaseScheduler<GeneratedDateQueueInDTO> {
+  constructor(
+    @InjectQueue(QueueNames.ECEProcessIntegration)
+    schedulerQueue: Queue<GeneratedDateQueueInDTO>,
+    private readonly eceRequest: ECEFileService,
+    queueService: QueueService,
+  ) {
+    super(schedulerQueue, queueService);
+  }
+
+  /**
+   * Identifies all the applications that have an eligible COE request waiting
+   * for a particular institution and generate the request file.
+   * @params job has generatedDate Date in which the assessment for
+   * particular institution is generated.
+   * @returns Processing result log.
+   */
+  @Process()
+  async processECERequest(
+    job: Job<GeneratedDateQueueInDTO | undefined>,
+  ): Promise<ECEProcessResultQueueOutDTO[]> {
+    this.logger.log(
+      `Processing ECE request integration job ${job.id} of type ${job.name}.`,
+    );
+    this.logger.log("Executing ECE request file generation ...");
+    const uploadResult = await this.eceRequest.processECEFile(
+      job.data.generatedDate,
```

Comment:
We need to override the `payload()` from the base class for the job data to be sent to the scheduler.

It will send undefined to job data if we do not overrride.

![image](https://user-images.githubusercontent.com/54600590/210865846-e30fa2a1-6793-48c4-8e59-c9665ab46802.png)

e.g. 
```
protected async payload(): Promise<GeneratedDateQueueInDTO> {
    return { generatedDate:someDate };
  }
```